### PR TITLE
Fix the S3 alternate runtime retries test in orchestrator mode

### PR DESCRIPTION
**Pull request by jdisanti**
This PR fixes the S3 `alternative-async-runtime` retry tests. The retry backoff time was being included in the operation attempt timeout, which I think is undesirable as it makes retry config much harder to get right since the backoff times are not predictable (they have randomness incorporated into them). The overall operation timeout is the only one that needs to incorporate backoff times.

In addition, this PR also:
- Updates READMEs for the `aws-smithy-runtime-api` and `aws-smithy-runtime` crates
- Adds top-level crate docs to describe features to `aws-smithy-runtime`
- Copies `capture_test_logs` into `aws-smithy-runtime` so that it can be used (just about) everywhere instead of just in `aws-config`
- Adds service/operation name to the tracing `invoke` span so it's possible to tell which operation the events are for
- Makes the `Debug` impl for `Identity` useful
- Adds a ton of trace/debug spans and events to the orchestrator
- Fixes an issue in `aws-smithy-runtime` where a huge amount of the orchestrator tests weren't being compiled due to a removed feature flag

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: rust-runtime/aws-smithy-runtime-api/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# aws-smithy-retries
+# aws-smithy-runtime-api
```
Comment (Velfi) [inline rust-runtime/aws-smithy-runtime-api/README.md:1]: lol I'm surprised it took us so long to notice this one.

File: rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs [hunk 2]
```diff
@@ -138,12 +159,15 @@ async fn try_op(
     stop_point: StopPoint,
 ) {
     // Before serialization
+    trace!("running read_before_serialization interceptors");
     halt_on_err!([ctx] => interceptors.read_before_serialization(ctx, cfg));
+    trace!("running modify_before_serialization interceptors");
     halt_on_err!([ctx] => interceptors.modify_before_serialization(ctx, cfg));
```
Comment (Velfi) [inline rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs:165]: Consider calling `trace` from the `Interceptors` thing. That way, there's less lines in the invoke methods.

File: rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs [hunk 3]
```diff
@@ -196,23 +223,25 @@ async fn try_op(
...
+        }
         let attempt_timeout_config = cfg.maybe_timeout_config(TimeoutKind::OperationAttempt);
+        trace!(attempt_timeout_config = ?attempt_timeout_config);
         let maybe_timeout = async {
-            // We must await this here or else timeouts won't work as expected
-            if let Some(delay) = retry_delay.take() {
-                delay.await;
-            }
-
+            debug!("beginning attempt #{i}");
```
Comment (Velfi) [inline rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs:244]: Thanks for fixing this. I really wasn't sure that I'd gotten it right.

**Review by Velfi — APPROVED** [review]

File: rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs (hunk 2 above)
Comment (jdisanti) [reply rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs:165]: Good idea...

**Review by Velfi — APPROVED** [review]

**Pull request merged**